### PR TITLE
modified /mmcv/runner/hooks/lr_updater.py: the 121 line

### PR DESCRIPTION
**Pull request by lws6885**
## Motivation

Learning rate did not update while trainning from a checkpoint file. So, i debugged and found that codes modified did not work.

## Modification

1、Commented the code line:"group.setdefault('initial_lr', group['lr'])" in line 121.
2、Added the code line:"group['initial_lr'] = group['lr']"

### CODE REVIEW

File: mmcv/runner/hooks/lr_updater.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def before_run(self, runner):
                 self.base_lr.update({k: _base_lr})
         else:
             for group in runner.optimizer.param_groups:
-                group.setdefault('initial_lr', group['lr'])
+                # group.setdefault('initial_lr', group['lr'])
+                group['initial_lr'] = group['lr']
```
Comment (zhouzaida) [inline mmcv/runner/hooks/lr_updater.py:122]: When resuming from a checkpoint, if 'initial_lr' is not saved, it will be set according to the optimizer params. So if we directly set the `initial_lr` as `lr`, I think it may not be an expected behavior.

**Comment by lws6885** [comment]
When training from a checkpiont file, lr should be updated,right? But it not happened.







At 2021-11-19 22:20:03, "Zaida Zhou" ***@***.***> wrote:

@zhouzaida commented on this pull request.

In mmcv/runner/hooks/lr_updater.py:

> @@ -118,7 +118,8 @@ def before_run(self, runner):
                 self.base_lr.update({k: _base_lr})
         else:
             for group in runner.optimizer.param_groups:
-                group.setdefault('initial_lr', group['lr'])
+                # group.setdefault('initial_lr', group['lr'])
+                group['initial_lr'] = group['lr']


When resuming from a checkpoint, if 'initial_lr' is not saved, it will be set according to the optimizer params. So if we directly set the initial_lr as lr, I think it may not be an expected behavior.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhouzaida** [comment]
> initial_lr

`initial_lr` is the learning rate at the beginning of training, when we resume from a checkpoint, the expected value of `inital_lr` should also be the learning rate at the beginning of training instead of the learning rate at the current epoch.

BTW, you can refer to https://github.com/open-mmlab/mmclassification/blob/master/tools/visualizations/vis_lr.py to visualize the changing curve of learning rate.

**Comment by lws6885** [comment]
Thanks.


| |
刘文开
|
|
***@***.***
|
签名由网易邮箱大师定制
On 11/22/2021 00:34，Zaida ***@***.***> wrote：

initial_lr

initial_lr is the learning rate at the beginning of training, when we resume from a checkpoint, the expected value of inital_lr should also be the learning rate at the beginning of training instead of the learning rate at the current epoch.

BTW, you can refer to https://github.com/open-mmlab/mmclassification/blob/master/tools/visualizations/vis_lr.py to visualize the changing curve of learning rate.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zhouzaida** [comment]
> `initial_lr` is the learning rate at the beginning of training, when we resume from a checkpoint, the expected value of `inital_lr` should also be the learning rate at the beginning of training instead of the learning rate at the current epoch.
> 
> BTW, you can refer to https://github.com/open-mmlab/mmclassification/blob/master/tools/visualizations/vis_lr.py to visualize the changing curve of learning rate.

https://github.com/open-mmlab/mmdetection/issues/6572

**Comment by zhouzaida** [comment]
Hi @lws6885 , I close this PR, and if you have any questions feel free to reopen it.

**Comment by OpenMMLab-Assistant003** [comment]
Hi @lws6885！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @lws6885
